generate_map.py

```diff
@@ -0,0 +1,186 @@
+import random
+import math
+import matplotlib
+matplotlib.use('ps')
+import matplotlib.pyplot as plt
+from enum import Enum
+import numpy as np
+from scipy.interpolate import BSpline, splprep, splev
+import copy
+
+class Difficulty(Enum):
+    """Enum for Difficulty level"""
+    PATHETIC = 0
+    EASY = 1
+    MEDIUM = 2
+    HARD = 3
+    REALLY_HARD = 4
+    WTF = 5
+
+
+def intersect(A,B,C,D):
+    """Helper of is_crossing function
+    Returns True if line segments AB and CD intersect"""
+    ccw = lambda A,B,C: (C[1]-A[1]) * (B[0]-A[0]) > (B[1]-A[1]) * (C[0]-A[0])
+    return ccw(A,C,D) != ccw(B,C,D) and ccw(A,B,C) != ccw(A,B,D)
+
+
+def is_crossing(X,Y,i,j):
+    """Function that checks if two lines ((X[i],Y[i]), (X[i+1], Y[i+1])), ((X[j],Y[j]), (X[j+1],Y[j+1])) intersect"""
```

Comment:
Notacja z X,Y,i,j jest trochę myląca. Rozważałaś może dorzucenie jakiejś pomocniczej klasy punkt, linia itp?

Reply:
No owszem, przyszło mi to do głowy, jak już troche napisałam tego wszystkiego, ale uznałam, że póki mam flow to zostawię to na później, w końcu zapomniałam :) 
Pozmieniam to, to dobry pomysł, będzie bardziej czytelne